File: 448-find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.go
func findDisappearedNumbers(nums []int) []int {
    numSet := make(map[int]bool)
    for _, num := range nums {
        numSet[num] = true
    }

    missingNum := []int{}
    for i := 1; i <= len(nums); i++ {
        if _, found := numSet[i]; !found {
            missingNum = append(missingNum, i)
        }
    }

    return missingNum
}

File: 448-find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        uniqueNums = set(nums)
        res = []

        for i in range(1, len(nums)+1):
            if i not in uniqueNums:
                res.append(i)

        return res
        